factcast-store/src/main/java/org/factcast/store/registry/transformation/FactTransformerServiceImpl.java

```diff
@@ -49,6 +51,23 @@ public class FactTransformerServiceImpl implements FactTransformerService {
 
   @NonNull private final RegistryMetrics registryMetrics;
 
+  @NonNull private final StoreConfigurationProperties props;
+  private final ExecutorService pool;
+
+  public FactTransformerServiceImpl(
+      @NonNull TransformationChains chains,
+      @NonNull Transformer trans,
+      @NonNull TransformationCache cache,
+      @NonNull RegistryMetrics registryMetrics,
+      @NonNull StoreConfigurationProperties props) {
+    this.chains = chains;
+    this.trans = trans;
+    this.cache = cache;
+    this.registryMetrics = registryMetrics;
+    this.props = props;
+    this.pool = Executors.newFixedThreadPool(props.getSizeOfThreadPoolForBufferedTransformations());
```

Comment:
In the old solution we had a work stealing pool and Im pretty sure we had discussion abotu why but I cannot recall anymore :D

Nevertheless, I think it would be nice to have this pool monitored + a shutdownhook

```
    this.executorService =
        metrics.monitor(
           Executors.newWorkStealingPool(props.getSizeOfThreadPoolForBufferedTransformations()),
            "fetching-catchup");
```

Reply:
this is neither fetching, nor catchup :laughing: